meta-buildpacks/java/README.md

```diff
@@ -1 +1,183 @@
+it will install Maven and use that instead.
+
+The local Maven repository (`.m2` folder) will be cached between builds for dependency resolution. However, neither
+the `mvn` executable (if installed) nor the `.m2` folder will be available in your container at runtime.
+
+### How it works: Gradle
+The buildpack will use the [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) in your app's
+root directory to execute the build defined by your `build.gradle` and download your dependencies. Gradle's
+[dependency cache](https://docs.gradle.org/current/userguide/dependency_resolution.html#sec:dependency_cache) is
+persisted between builds to speed up subsequent builds. However, the dependency cache will not be available in your
+container at runtime.
+
+## Configuration
+### Procfile
+A Procfile is a text file in the root directory of your application that defines process types and explicitly declares
+what command should be executed to start your app. Your `Procfile` will look something like this for Spring Boot:
+
+```
+web: java -Dserver.port=$PORT $JAVA_OPTS -jar target/demo-0.0.1-SNAPSHOT.jar
+```
+
+### OpenJDK
+#### OpenJDK Version
+By default, the latest OpenJDK 8 release will be installed. You can configure the OpenJDK version your application
+needs. The buildpack tries to determine the required version in the following order:
+
+##### `system.properties`
+You can specify a Java version by adding a [Java properties file](https://en.wikipedia.org/wiki/.properties) called
+`system.properties` to the root directory of your application. The value of the `java.runtime.version` key specifies
+the required OpenJDK version:
+
+```
+java.runtime.version=15
+```
+
+Supported major versions are `1.7`, `1.8`, `11`, `13`, and `15`. The buildpack will always install the latest release
+of the requested major version.
+
+##### Environment variable: `BP_JVM_VERSION`
+You can use the same major version strings as in the `system.properties` file. In addition, it is allowed to append `.*`
+to a major version (i.e. `11.*` for OpenJDK 11). This ensures compatibility with the
+[Spring Boot Plugin](https://docs.spring.io/spring-boot/docs/2.4.1/maven-plugin/reference/htmlsingle/#goals-build-image).
+
+#### Customizing the JDK
+There are some cases where files need to be bundled with the JDK in order to expose functionality in the runtime JVM.
+For example, the inclusion of a custom certificate authority (CA) store is common. To handle such cases, this buildpack
+will copy files designated by the app in a `.jdk-overlay` folder into the JDK's directory structure.
+
+##### Adding custom certificates
+You may also need to add custom certificates to the JDK’s cacerts. You may start with the keystore in your local JDK or
+[download the base Heroku keystore](https://heroku-cacerts.herokuapp.com/heroku_cacerts). Add the custom certificate
```

Comment:
What team owns the `heroku-cacerts` app? The CA certs can vary by stack, so the app (a) should always use an up to date stack, (b) perhaps there may be times when the customer needs the castore associated with the stack their app is running, rather than that of `heroku-cacerts` at the time?